### DISCUSSION TITLE
Small change to git-commit-insert-header

### DISCUSSION THREAD

**Discussion by bigodel** [magit/magit#4811] [Ideas]
I have a custom `.gitmessage` with some helpful tips for writing meaningful commits in comments. It looks something like this:

```
# A brief resume of the changes. Keep it short!   |

# What changed. Don't describe in details!                              |

# Why it was changed.                                                   |
```

But because of that `git-commit-insert-header` inserts the header at the top of the buffer, since it is going to search of the first line containing a comment with the following code:

```lisp
(while (re-search-backward (concat "^" comment-start) nil t))
```

I am proposing to change it like so:

```diff
diff -u git-commit.el git-commit-new.el
--- git-commit.el	2022-11-08 15:14:24.018391274 -0300
+++ git-commit-new.el	2022-11-08 15:14:18.378386009 -0300
@@ -918,7 +918,9 @@
            (unless (= (char-after) ?\n)
              (insert ?\n)))
           (t
-           (while (re-search-backward (concat "^" comment-start) nil t))
+           (while (and
+                   (not (save-excursion (forward-line -1) (looking-at "^$")))
+                   (re-search-backward (concat "^" comment-start) nil t)))
            (unless (looking-back "\n\n" nil)
              (insert ?\n))
            (insert header ?\n)))
```

**Comment by bigodel** [comment]
Bump.

**Reply by tarsius** [reply]
That doesn't look right (didn't try). A solution that begins at eob and then uses `(forward-comment -1)` would probably be more robust. Of course there is the special case when there is no "official" comment at the end and the last comment from `.gitmessage` is not followed by any non-whitespace.
